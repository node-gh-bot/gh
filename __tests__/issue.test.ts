/**
 * Â© 2013 Liferay, Inc. <https://liferay.com> and Node GH contributors
 * (see file: CONTRIBUTORS)
 * SPDX-License-Identifier: BSD-3-Clause
 */

import { runCmd } from './runCmd'

describe('E2E: Issues Module Test', () => {
    it('List Issues `gh is`', done => {
        expect(runCmd('gh is')).toMatchSnapshot()
        done()
    })

    it(`Assign Issues \`gh is --assign -A ${process.env.GH_USER} --number 1\``, done => {
        expect(runCmd(`gh is --assign -A ${process.env.GH_USER} --number 1`)).toMatchSnapshot()
        done()
    })

    it('Create a new issue `gh is -N -t "Node GH rocks!" -L bug,question,test`', done => {
        expect(runCmd(`gh is -N -t "Node GH rocks!" -L bug,question,test`)).toMatchSnapshot()
        done()
    })

    it('Comment on Issues `gh is 1 -c "comment"`', done => {
        expect(runCmd(`gh is 1 -c "comment"`)).toMatchSnapshot()
        done()
    })

    it('Open Issue `gh is 1 -o`', done => {
        expect(runCmd(`gh is 1 -o`)).toMatchSnapshot()
        done()
    })

    it('Close Issue `gh is 1 -C`', done => {
        expect(runCmd(`gh is 1 -C`)).toMatchSnapshot()
        done()
    })

    it('Search Issues `gh is -s hi`', done => {
        expect(runCmd(`gh is -s 'hi'`)).toMatchSnapshot()
        done()
    })
})
