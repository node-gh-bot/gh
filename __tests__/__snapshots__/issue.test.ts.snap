// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E: Issues Module Test Assign Issues \`gh is --assign -A protoEvangelion --number 1\` 1`] = `
"Assigning issue #1 on protoEvangelion/gh to protoEvangelion
https://github.com/protoEvangelion/gh/issues/1
"
`;

exports[`E2E: Issues Module Test Close Issue \`gh is 1 -C\` 1`] = `
"Closing issue #1 on protoEvangelion/gh
https://github.com/protoEvangelion/gh/issues/1
"
`;

exports[`E2E: Issues Module Test Comment on Issues \`gh is 1 -c "comment"\` 1`] = `
"Adding comment on issue #1 on protoEvangelion/gh
https://github.com/protoEvangelion/gh/issues/1#issuecomment-000000001
"
`;

exports[`E2E: Issues Module Test Create a new issue \`gh is -N -t "Node GH rocks!" -L bug,question,test\` 1`] = `
"Creating a new issue on protoEvangelion/gh
https://github.com/protoEvangelion/gh/issues/88
[hook] gh is --browser  --user protoEvangelion --repo gh --number 88
"
`;

exports[`E2E: Issues Module Test List Issues \`gh is\` 1`] = `
"Listing open issues on protoEvangelion/gh
#77 'Node GH rocks!' @protoEvangelion (in 4 years)

#76 test2 @protoEvangelion (in 4 years)

#74 test @protoEvangelion (in 4 years)
"
`;

exports[`E2E: Issues Module Test Open Issue \`gh is 1 -o\` 1`] = `
"Opening issue #1 on protoEvangelion/gh
https://github.com/protoEvangelion/gh/issues/1
"
`;

exports[`E2E: Issues Module Test Search Issues \`gh is -s hi\` 1`] = `
"Searching for hi in issues for protoEvangelion/gh

#72 hi @protoEvangelion (in 4 years)

#65 hi @protoEvangelion (in 4 years)

#63 hi @protoEvangelion (in 4 years)

#62 hi @protoEvangelion (in 4 years)

#61 hi @protoEvangelion (in 4 years)

#60 hi @protoEvangelion (in 4 years)

#19 hi @protoEvangelion (in 4 years)

#1 test1 @protoEvangelion (in 4 years)
"
`;
